examples/data-objects/datastore-race/src/index.tsx

```diff
@@ -0,0 +1,252 @@
+import {
+    IFluidDataStoreContext,
+    IFluidDataStoreFactory,
+ } from "@fluidframework/runtime-definitions";
+import { IFluidHTMLOptions, IFluidHTMLView } from "@fluidframework/view-interfaces";
+import React from "react";
+import ReactDOM from "react-dom";
+import { FluidDataStoreRuntime, FluidObjectHandle, ISharedObjectRegistry } from "@fluidframework/datastore";
+import { SharedMap } from "@fluidframework/map";
+import {
+    IChannelFactory,
+} from "@fluidframework/datastore-definitions";
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+import { IRuntimeFactory, IContainerContext } from "@fluidframework/container-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { RequestParser } from "@fluidframework/runtime-utils";
+
+const counterKey = "counter";
+
+export class MergeableClicker implements IFluidHTMLView, IFluidRouter, IFluidLoadable {
+    public static async load(runtime: FluidDataStoreRuntime) {
+        const ds = new MergeableClicker(runtime);
+        if (runtime.existing === false) {
+            await ds.initializingFirstTime();
+        }
+        await ds.hasInitialized();
+        return ds;
+    }
+
+    public get IFluidHTMLView() { return this; }
+    public get IFluidRouter() { return this; }
+    public get IFluidLoadable() { return this; }
+
+    public readonly handle: IFluidHandle<this>;
+
+    private root!: SharedMap;
+    private _counter: SharedCounter | undefined;
+
+    private constructor(private readonly runtime: FluidDataStoreRuntime) {
+        this.handle = new FluidObjectHandle(this, "", runtime.objectsRoutingContext);
+    }
+    async request(request: IRequest): Promise<IResponse> {
+        return {
+            mimeType: "fluid/object",
+            status: 200,
+            value: this,
+        };
+    }
+
+    public async initializingFirstTime() {
+        const root = SharedMap.create(this.runtime, "root");
+        const counter = SharedCounter.create(this.runtime);
+        root.set(counterKey, counter.handle);
+        root.bindToContext();
+    }
+
+    public async hasInitialized() {
+        this.root = await this.runtime.getChannel("root") as unknown as SharedMap;
+        const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
+        this._counter = await counterHandle.get();
+        console.log(`counter ${this._counter?.value}`);
+    }
+
+    public render(div: HTMLElement) {
+        // Get our counter object that we set in initialize and pass it in to the view.
+        ReactDOM.render(
+            <CounterReactView counter={this.counter} />,
+            div,
+        );
+        return div;
+    }
+
+    private get counter() {
+        if (this._counter === undefined) {
+            throw new Error("SharedCounter not initialized");
+        }
+        return this._counter;
+    }
+
+    public merge(target: this) {
+        target.counter.increment(this.counter.value);
```

Comment:
right now we just race at the datastore level, and the datastore needs to handle the full merge. We could likey develope dds specific helpers to ease merging.